projects/plugins/jetpack/modules/sso.php

```diff
@@ -98,6 +166,143 @@ public static function get_instance() {
 		return self::$instance;
 	}
 
+	/**
+	 * Adds a "blank" column in the user admin table to display indication of user connection.
+	 *
+	 * @param array $columns User list table columns.
+	 *
+	 * @return array
+	 */
+	public function jetpack_user_connected_th( $columns ) {
+		$columns['user_jetpack'] = sprintf(
+			'<span title="%1$s">[?]</span>',
+			esc_attr__( 'Connected users can log-in to this site using their WordPress.com account.', 'jetpack' )
+		);
+		return $columns;
+	}
+
+	/**
+	 * Check if a given user is invited to the site.
+	 *
+	 * @access private
+	 * @static
+	 * @param int $user_id The user ID.
+	 *
+	 * @return array An array of capabilities
+	 */
+	private static function get_is_user_invited( $user_id ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$blog_id = Jetpack_Options::get_option( 'id' );
+
+		$user     = get_user_by( 'id', $user_id );
+		$url      = '/sites/' . $blog_id . '/invites/is-invited';
+		$url      = add_query_arg(
+			array(
+				'email_or_username' => $user->user_email,
+			),
+			$url
+		);
+		$response = Client::wpcom_json_api_request_as_user(
+			$url,
+			'v2',
+			array(),
+			null,
+			'wpcom'
+		);
+		if ( is_wp_error( $response ) ) {
+			return false;
+		}
+
+		if ( 200 !== $response['response']['code'] ) {
+			return false;
+		}
+
+		return json_decode( $response['body'], true )['slug'];
+	}
+
+	/**
+	 * Show Jetpack icon if the user is linked.
+	 *
+	 * @param string $val HTML for the icon.
+	 * @param string $col User list table column.
+	 * @param int    $user_id User ID.
+	 *
+	 * @return string
+	 */
+	public function jetpack_show_connection_status( $val, $col, $user_id ) {
+		if ( 'user_jetpack' === $col && Jetpack::connection()->is_user_connected( $user_id ) ) {
+			$connection_html = sprintf(
+				'<button disabled title="%1$s" class="jetpack-sso-invitation">%2$s</button>',
+				esc_attr__( 'This user is connected and can log-in to this site.', 'jetpack' ),
+				esc_attr__( 'Connected', 'jetpack' )
+			);
+			return $connection_html;
+		} else {
+			$has_pending_invite = self::get_is_user_invited( $user_id );
+			if ( $has_pending_invite ) {
+				$connection_html = sprintf(
+					'<button disabled title="%1$s" class="jetpack-sso-invitation sso-pending-invite">%2$s</button>',
+					esc_attr__( 'This user didn\'t accept the invitation to join this site yet.', 'jetpack' ),
```

Comment:
```suggestion
					esc_attr__( 'This user didn&#8217;t accept the invitation to join this site yet.', 'jetpack' ),
```
[WordPress solves this](https://github.com/search?q=repo%3AWordPress%2FWordPress%208217&type=code) by using the html entity of a curly single quote.